File: ResourceManager/Models/Configurations/PermissionConfiguration.vb
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Metadata.Builders
Imports ResourceManager.Configuration

Namespace ResourceManager.Models.Configurations
    Public Class PermissionConfiguration
        Implements IEntityTypeConfiguration(Of Permission)
        Public Sub Configure(ByVal builder As EntityTypeBuilder(Of Permission)) Implements IEntityTypeConfiguration(Of Permission).Configure
            builder.ToTable("Permissions")
            builder.HasKey(Function(x) x.ID)

            builder.Property(Function(x) x.ID).ValueGeneratedOnAdd()
            builder.[Property](Function(x) x.CreatedDate).HasDefaultValueSql("date('now','localtime')")
            builder.[Property](Function(x) x.ModifiedDate).HasDefaultValueSql("date('now','localtime')")
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Name), "nvarchar(100)").IsRequired()
            RelationalPropertyBuilderExtensions.HasColumnType(builder.Property(Function(x) x.Description), "nvarchar(255)").IsRequired()
            builder.HasIndex(CType(Function(x) x.Name, Expressions.Expression(Of Func(Of Permission, Object)))).IsUnique()

            builder.HasData(DesignData.Instance.Permissions)
        End Sub
    End Class
End Namespace

File: ResourceManager/Models/Configurations/PermissionConfiguration.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using ResourceManager.Configuration;
using System.Collections.Generic;

namespace ResourceManager.Models.Configurations
{
    public class PermissionConfiguration : IEntityTypeConfiguration<Permission>
    {
        public void Configure(EntityTypeBuilder<Permission> builder)
        {
            builder.ToTable("Permissions");
            builder.HasKey(x => x.ID);

            builder.Property(x => x.ID).ValueGeneratedOnAdd();
            builder.Property(x => x.CreatedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.ModifiedDate).HasDefaultValueSql("date('now','localtime')");
            builder.Property(x => x.Name).HasColumnType("nvarchar(100)").IsRequired();
            builder.Property(x => x.Description).HasColumnType("nvarchar(255)").IsRequired();
            builder.HasIndex(x => x.Name).IsUnique();

            builder.HasData(DesignData.Instance.Permissions);
        }
    }
}
